src/som/vmobjects/SInteger.java

```diff
@@ -24,19 +25,38 @@
 
 package som.vmobjects;
 
-import java.math.BigInteger;
-
 import som.vm.Universe;
 
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.Map;
+
 public final class SInteger extends SNumber {
 
+  /**
+   * Language convention requires integers up to this value to be identical.
+   */
+  private static final long MAX_IDENTICAL_INT = 1073741823L;
```

Comment:
Ok, just checked, I guess that's the very first test for integers `testEqualityAndIdentity`.
Not exactly sure how you picked `1073741823L`.
I wonder how well this works on the benchmarks though. Iterating over large ranges of integers might be pretty expensive because of the hash access.